src/services/AllSitesService.ts

```diff
@@ -1,8 +1,15 @@
-import { SiteDataRequest } from "types/sites"
+import { SiteDataRequest, SitePreviewRequest } from "types/sites"
 
 import { apiService } from "./ApiService"
 
 export const getAllSites = async (): Promise<SiteDataRequest> => {
   const endpoint = `/sites`
   return apiService.get(endpoint).then((res) => res.data)
 }
+
+export const getSitePreview = async (
+  sites: string[]
+): Promise<SitePreviewRequest[]> => {
+  const endpoint = `/sites/preview`
+  return apiService.post(endpoint, { sites }).then((res) => res.data)
```

Comment:
is there a reason why this is a post and not a get ah?

Reply:
I didn't find a way to send a list of strings (sites) through a get request, and therefore resorted to a post request. If you know of a way pls lmk

Reply:
hmm why cant we do a 

get request on `/sites/preview`, and return only a list of sites that 
1. has an image that is defined 
2. if admin return empty array so this wouldnt slow down our network calls

our cookies already has the user info, so the FE does not need to compute this

Reply:
Can I check if you're suggesting to get list of permitted sites via DB? I think that would result in Github users not being able to get images thou